digitalocean/resource_digitalocean_project.go

```diff
@@ -0,0 +1,291 @@
+	client := meta.(*CombinedConfig).godoClient()
+
+	project, resp, err := client.Projects.Get(context.Background(), d.Id())
+
+	if err != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			log.Printf("[DEBUG] Project  (%s) was not found - removing from state", d.Id())
+			d.SetId("")
+		}
+
+		return fmt.Errorf("Error reading Project: %s", err)
+	}
+
+	d.Set("id", project.ID)
+	d.Set("name", project.Name)
+	d.Set("purpose", strings.TrimPrefix(project.Purpose, "Other: "))
+	d.Set("description", project.Description)
+	d.Set("environment", project.Environment)
+	d.Set("is_default", project.IsDefault)
+	d.Set("owner_uuid", project.OwnerUUID)
+	d.Set("owner_id", project.OwnerID)
+	d.Set("created_at", project.CreatedAt)
+	d.Set("updated_at", project.UpdatedAt)
+
+	urns, err := loadResourceURNs(client, project.ID)
+	if err != nil {
+		return fmt.Errorf("Error reading Project: %s", err)
+	}
+
+	d.Set("resources", urns)
+
+	return err
+}
+
+func resourceDigitalOceanProjectUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*CombinedConfig).godoClient()
+	projectId := d.Id()
+
+	d.Partial(true)
+
+	projectRequest := &godo.UpdateProjectRequest{
+		Name:        d.Get("name"),
+		Description: d.Get("description"),
+		Purpose:     d.Get("purpose"),
+		Environment: d.Get("environment"),
+		IsDefault:   d.Get("is_default"),
+	}
+
+	_, _, err := client.Projects.Update(context.Background(), projectId, projectRequest)
+
+	if err != nil {
+		return fmt.Errorf("Error updating Project: %s", err)
+	}
+
+	d.SetPartial("project_updated")
+
+	// The API requires project resources to be reassigned to another project if the association needs to be deleted.
+	// a diff of the resource could be implemented instead of removing all, (bulk) and adding the back again.
+	if d.HasChange("resources") {
+		oldURNs, newURNs := d.GetChange("resources")
+
+		assignResourcesToDefaultProject(client, oldURNs.(*schema.Set))
+
+		var urns *[]interface{}
+
+		if newURNs.(*schema.Set).Len() != 0 {
+			urns, err = assignResourcesToProject(client, projectId, newURNs.(*schema.Set))
+			if err != nil {
+				return fmt.Errorf("Error Updating project: %s", err)
+			}
+		}
+
+		d.Set("resources", urns)
+		d.SetPartial("project_resources_updated")
+	}
+
+	log.Printf("[INFO] Updated Project, ID: %s", projectId)
+	d.Partial(false)
+
+	return resourceDigitalOceanProjectRead(d, meta)
+}
+
+func resourceDigitalOceanProjectDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*CombinedConfig).godoClient()
+
+	defaultProject, _, defaultProjErr := client.Projects.GetDefault(context.Background())
+	if defaultProjErr != nil {
+		return fmt.Errorf("Error locating default project %s", defaultProjErr)
+	}
+
+	projectId := d.Id()
+	defaultProjectId := defaultProject.ID
+
+	log.Printf("[DEBUG] Default project located, ID: %s", defaultProjectId)
+
+	if v, ok := d.GetOk("resources"); ok {
+
+		_, err := assignResourcesToProject(client, defaultProjectId, v.(*schema.Set))
```

Comment:
Seem like we could use `assignResourcesToDefaultProject` to reduce some code duplication here.